docs/reference/indexes.txt

```diff
@@ -82,6 +82,18 @@ Indexes can be scoped to a specific database:
       index({ ssn: 1 }, { database: "users", unique: true, background: true })
     end
 
+You may use aliased field names in index definitions. Field aliases
+will also be resolved on the following options: ``partial_filter_expression``,
```

Comment:
The example is using storage field name rather than a field alias right?

Reply:
In the example:

```ruby
      field :a, as: :age
      index({ age: 1 }, { partial_filter_expression: { age: { '$gte' => 20 } })
```

`:a` is the storage field name, `:age` is the alias, so the example illustrates how to use the alias in the Ruby index definition.

Reply:
Aliases are defined in https://www.mongodb.com/docs/mongoid/master/reference/fields/#field-aliases. Are they supported in the implementation added by this PR?

Reply:
Yes, because they are returned by `database_field_name`

```ruby
class Band
  include Mongoid::Document
  field :name, type: String
  alias_attribute :n, :name
end

Band.database_field_name(:n) #=> "name"
```

Reply:
Storage field names and field aliases are separate features, if both are supported they need to be both mentioned in the docs and there need to be test cases for each of them.

Reply:
I'll do it, but bear in mind we have 1000s of tests for alias-related features today, and *literally none* of them test the case of `alias_attribute` being different than `:as`. The goal post seems to constantly be moving here.

Reply:
Specs for `alias_attribute` cases are added.